packages/website/ts/pages/governance/modal_vote.tsx

```diff
@@ -0,0 +1,304 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import styled from 'styled-components';
+
+import { colors } from 'ts/style/colors';
+
+import { DialogContent, DialogOverlay } from '@reach/dialog';
+import '@reach/dialog/styles.css';
+
+import { ContractWrappers } from '@0x/contract-wrappers';
+import { BigNumber } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import { Button } from 'ts/components/button';
+import { Icon } from 'ts/components/icon';
+import { Heading, Paragraph } from 'ts/components/text';
+import { GlobalStyle } from 'ts/constants/globalStyle';
+
+import { ConnectForm, WalletConnectedProps } from 'ts/pages/governance/connect_form';
+import { ErrorModal } from 'ts/pages/governance/error_modal';
+import { VoteForm, VoteInfo } from 'ts/pages/governance/vote_form';
+
+import { LedgerSubprovider } from '@0x/subproviders';
+import { Provider } from 'ethereum-types';
+import { configs } from 'ts/utils/configs';
+import { constants } from 'ts/utils/constants';
+
+interface Props {
+    theme?: GlobalStyle;
+    isOpen?: boolean;
+    onDismiss?: () => void;
+    onWalletConnected?: (providerName: string) => void;
+    onVoted?: (voteInfo: VoteInfo) => void;
+}
+
+interface State {
+    currentBalance: BigNumber;
+    isWalletConnected: boolean;
+    providerName: string;
+    isSubmitting: boolean;
+    isLedger: boolean;
+    ledgerSubproviderIfExists?: LedgerSubprovider;
+    isSuccessful: boolean;
+    isErrorModalOpen?: boolean;
+    isU2fSupported: boolean;
+    isVoted: boolean;
+    votePreference: string | null;
+    zeip: number;
+    voteHash?: string;
+    signedVote?: any;
+    errorMessage?: string;
+    errors: ErrorProps;
+    web3Wrapper?: Web3Wrapper;
+    contractWrappers?: ContractWrappers;
+    providerEngine?: Provider;
+    web3?: any;
+    selectedAddress?: string;
+}
+
+interface FormProps {
+    isSuccessful?: boolean;
+    isSubmitting?: boolean;
+}
+
+interface ErrorProps {
+    [key: string]: string;
+}
+
+export class ModalVote extends React.Component<Props> {
+    public networkId: number;
+    public state: State = {
+        currentBalance: new BigNumber(0),
+        isWalletConnected: false,
+        isU2fSupported: false,
+        providerName: 'Metamask',
+        selectedAddress: null,
+        isSubmitting: false,
+        isLedger: false,
+        ledgerSubproviderIfExists: null,
+        providerEngine: null,
+        isSuccessful: false,
+        isVoted: false,
+        votePreference: null,
+        zeip: 23,
+        errors: {},
+    };
+    // shared fields
```

Comment:
lets put a comment here about how this is copy and pasted from the generic form and includes a lot of extraneous stuff (i.e. MM lead fields)